src/components/storage/CockpitStorageIntegration.jsx

```diff
@@ -0,0 +1,222 @@
+    scenarioPartitioningMapping,
+    selectedDisks,
+    setShowStorage,
+    setStorageScenarioId,
+}) => {
+    return (
+        <>
+            <PageSection
+              stickyOnBreakpoint={{ default: "top" }}
+              variant={PageSectionVariants.light}
+            >
+                <Flex spaceItems={{ default: "spaceItemsLg" }}>
+                    <Title headingLevel="h1" size="2xl">{_("Configure storage")}</Title>
+                    <Alert
+                      isInline
+                      isPlain
+                      title={_("Changes made here will immediately affect the system. There is no 'undo'.")}
+                      variant="warning"
+                    />
+                </Flex>
+            </PageSection>
+            <div className={idPrefix + "-page-section-cockpit-storage"}>
+                <PageSection>
+                    <iframe
+                      src="/cockpit/@localhost/storage/index.html"
+                      name="cockpit-storage"
+                      className={idPrefix + "-iframe-cockpit-storage"} />
+                </PageSection>
+                <ModifyStorageSideBar />
+            </div>
+            <PageSection
+              className={idPrefix + "-page-section-storage-footer"}
+              stickyOnBreakpoint={{ default: "bottom" }}
+              variant={PageSectionVariants.light}
+            >
+                <ReturnToInstallationButton
+                  dispatch={dispatch}
+                  onCritFail={onCritFail}
+                  setShowStorage={setShowStorage}
+                />
+            </PageSection>
+        </>
+    );
+};
+
+const ModifyStorageSideBar = () => {
+    const mountPointConstraints = useMountPointConstraints();
+    const requiredSize = useRequiredSize();
+
+    if (mountPointConstraints === undefined) {
+        return null;
+    }
+
+    const requiredConstraints = (
+        mountPointConstraints.filter(constraint => constraint.required.v)
+    );
+    const recommendedConstraints = (
+        mountPointConstraints.filter(constraint => !constraint.required.v && constraint.recommended.v)
+    );
+
+    const requiredConstraintsSection = (
+        requiredConstraints.length > 0 &&
+        <>
+            <Text component="p" className={idPrefix + "-requirements-hint"}>
+                {_("If you are configuring partitions the following are required:")}
+            </Text>
+            <List className={idPrefix + "-requirements-hint-list"}>
+                {requiredConstraints.map(constraint => {
+                    const item = [
+                        constraint["mount-point"].v,
+                        constraint["required-filesystem-type"].v
+                    ]
+                            .filter(c => !!c)
+                            .join(" ");
+
+                    return <ListItem key={item}>{item}</ListItem>;
+                })}
+            </List>
+        </>
+    );
+    const recommendedConstraintsSection = (
```

Comment:
Maybe some part could be shared with `requiredConstraintsSection`, but it can be too early or not worthy given the potential follow-ups.

Reply:
Indeed - very easy to share, moved the code accordingly.